ntoskrnl/ke/amd64/thrdini.c

```diff
@@ -161,11 +161,7 @@ KiSwapContextResume(
     NewProcess = NewThread->ApcState.Process;
     if (OldProcess != NewProcess)
     {
-        /* Switch address space and flush TLB */
-        __writecr3(NewProcess->DirectoryTableBase[0]);
-
-        /* Set new TSS fields */
-        //Pcr->TssBase->IoMapBase = NewProcess->IopmOffset;
+        KiSwapProcess(NewProcess, OldProcess);
```

Comment:
Note: a similar change could be done in x86 in [`KiSwapContextExit`](https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/ke/i386/thrdini.c;hb=b1cf981c52f5d01774eca1b0dde77ee3fa3322a3#l341) (compare with the [x86 `KiSwapProcess` implementation](https://git.reactos.org/?p=reactos.git;a=blob;f=ntoskrnl/ke/i386/context.c;hb=b1cf981c52f5d01774eca1b0dde77ee3fa3322a3#l21)) -- this would ensure having the SMP-specific code run there as well.

Reply:
This is not an SMP PR and won't change anything in x86 that isn't strictly needed. I'll leave that part to @DarkFire01.